src/ActionCreator.js

```diff
@@ -6,9 +8,30 @@ let makeDeleteAction = (snap) => {
 
 // Given an EBS volume and a list of snapshots, return a list of actions
 // to create the necessary snapshots that fulful the backup requirements
-let makeCreationActions = (volume, snapList) => {
+let makeCreationActions = (volume) => {
+	let backupTypes = volume.BackupConfig.BackupTypes.filter(type => {
+		let now = moment();
+		if (!volume.Snapshots[type.Name] ||
+				volume.Snapshots[type.Name].length <= 0 ||
+				now.diff(volume.Snapshots[type.Name][0].StartTime, 'hours', true) > type.Frequency
+		) {
+			return true;
+		} else {
+			return false;
+		}
+	});
 
-	return determineBackupsNeeded(volume, snapList).map(backup => backup);
+	let actions = [];
+	backupTypes.map((backup) => {
```

Comment:
Why `map` and `push`? Why not just map?

Also, https://facebook.github.io/immutable-js/ A+